napari_plugin_engine/manager.py

```diff
@@ -236,12 +239,30 @@ def discover(
         for name, mod_name, dist_name in self.iter_available(
             path, entry_point, prefix
         ):
-            if self.is_registered(name) or self.is_blocked(name):
+            old_name = name
+            # different plugin has already registered this entry point
+            if self.is_registered(name):
+                mod_names = [
```

Comment:
might be slightly faster if you made this a generator (use `()` instead of list comprehension) ... as the `not in ___` expression will exit as soon as it hits a match

Reply:
thanks for the tip, that's done now :)